### ISSUE TITLE
Upgrade Datastore to use v1beta3

### ISSUE THREAD

**Issue by JustinBeckwith** [GoogleCloudPlatform/gcloud-ruby#314]

**Comment by blowmage** [comment]
Hey @JustinBeckwith, thanks for the issue! The Google Cloud [Datastore docs](https://cloud.google.com/datastore/docs/apis/v1beta2/datasets/lookup) are still showing `v1beta2`. Is `v1beta3` available? If not, do you know when it will be generally available?

**Comment by jgeewax** [comment]
@pcostell, can you link us to the v1beta3 docs?

**Comment by pcostell** [comment]
`v1beta3` is in alpha/pre-alpha(?). The protos live here: https://github.com/google/googleapis/tree/master/google/datastore/v1beta3. I can email anyone the release docs (we are waiting until v1beta3 is available for everyone to actually stage them). Also, let me know if you need an app whitelisted for testing.

**Comment by blowmage** [comment]
How long will v1beta2 be supported during the transition to v1beta3?

**Comment by pcostell** [comment]
Here are the release notes: https://cloud.google.com/datastore/docs/release-notes. 

We don't have any plans for the timeline for deprecating v1beta2. v1beta3 should be significantly more performant so it is probably good for users to transition as quickly as possible. Once the gcloud-* libraries move over we'll have a better sense of how long we should give people to transition.

**Comment by blowmage** [comment]
Great! So if we switch now will all users be able to use v1beta3? Is it generally available? Or will they need to request access?

**Comment by pcostell** [comment]
No it is not yet available. We'll update everyone as soon as it is. I can get Google developers early access, but we don't have a public early access program.

**Comment by blowmage** [comment]
@pcostell Can you allow access to v1beta3 to helical-zone-771 and nodal-almanac-725?

**Comment by pcostell** [comment]
You're all set.

**Comment by pcostell** [comment]
@blowmage any progress here?

**Comment by blowmage** [comment]
In discussions with @jgeewax we decided to hold off on upgrading to v1beta3 until it was publicly available. I hope this will be soon because I'd love to get this change made. :)

**Comment by blowmage** [comment]
The protobuf files for v1beta3 are available [here](https://github.com/google/googleapis/tree/master/google/datastore/v1beta3).

**Comment by pcostell** [comment]
What is left for this? The code is in a separate branch right now, what needs to happen to get it merged into master and deployed?

**Comment by quartzmo** [comment]
@pcostell Thanks for asking! We are blocked by google/protobuf#1204, a known issue in Google's Protobuf library's support of `oneof` used in the datastore proto3 definitions. The issue is supposed to be fixed in PR google/protobuf#1139, which has been merged into master, but has not yet been released. As you can see if you scroll down the comments in google/protobuf#1204, @blowmage has asked a couple of times about the release. We have also asked @jgeewax to look into it.
